lib/ipaddr.rb

```diff
@@ -149,7 +204,7 @@ def ~
   # Returns true if two ipaddrs are equal.
   def ==(other)
     other = coerce_other(other)
-    return @family == other.family && @addr == other.to_i
+    return @family == other.family && @addr == other.to_i && mask_addr == other.mask_addr
```

Comment:
I'm afraid this change is unacceptable.  Changing the sense of equality would break existing code.